src/main/java/net/sf/jabref/gui/PreviewPanel.java

```diff
@@ -275,16 +275,35 @@ public void setLayout(Layout layout) {
     }
 
     public void setEntry(BibEntry newEntry) {
-        if(entry.isPresent() && (entry.get() != newEntry)) {
-            entry.ifPresent(e -> e.removePropertyChangeListener(this));
-            newEntry.addPropertyChangeListener(this);
+
+        if (entry.isPresent() && (entry.get() != newEntry)) {
+            entry.ifPresent(e -> e.unregisterListener(this.currentChangeFieldUpdateEvent));
+            this.currentChangeFieldUpdateEvent = new ChangeFieldUpdateEvent();
+            newEntry.registerListener(this.currentChangeFieldUpdateEvent);
         }
+
         entry = Optional.ofNullable(newEntry);
 
         updateLayout();
         update();
     }
 
+
+    /**
+    * Listener class for ChangeFieldUpdateEvent.
```

Comment:
The name is confusing. Why is "Change" and "Update" in the name? Is it only "Change"? The event should be called `FieldAddedChangedRemovedEvent` and this listener `FieldAddedChangedRemovedEventListener` (meaning `Listener` added to the name of the event class).